client/src/legacy/GridField.js

```diff
@@ -180,6 +194,19 @@ $.entwine('ss', function($) {
       if (hasLegacyFilterHeader) {
         this.find('.sortable-header th:last').html(content);
       }
+    },
+
+    keepStateInHistory: function() {
+      const newURLSceme = $(this).find('.gridfield-actionmenu__container').data('schema');
+      if (newURLSceme && newURLSceme.length > 0) {
+        newURLSceme.filter( e => {
+          if (e.type === 'link') {
+            const searchParam = e.url.includes('?') ? '?' + e.url.split('?')[1] : '';
+            const historyState = $.extend({}, {path: window.location.pathname + searchParam}, this.getState());
+            history.replaceState(historyState, '', window.location.pathname + searchParam);
```

Comment:
This might be a good way to preserve existing URL parameters while adding the ones from the element:
```js
const params = [window.location.search, (new URL(e.url)).searchParams.toString()].join('&');
```

Reply:
DONE